pkg/controllers/cloud/networkpolicy_controller.go

```
@@ -127,8 +129,29 @@ func (r *NetworkPolicyReconciler) isNetworkPolicySupported(anp *antreanetworking
 	return nil
 }
 
+// updateRuleRealizationStatus check rule realization status on all appliedTo groups for a np and send status.
+func (r *NetworkPolicyReconciler) updateRuleRealizationStatus(np *networkPolicy, err error) {
+	if err == nil {
+		for _, at := range np.AppliedToGroups {
+			sgs, e := r.appliedToSGIndexer.ByIndex(addrAppliedToIndexerByGroupID, at)
+			if e != nil {
+				r.Log.Error(e, "get appliedToSG indexer", "sg", at)
+				return
+			}
+			for _, obj := range sgs {
+				sg := obj.(*appliedToSecurityGroup)
+				e = sg.checkRealization(r, np)
```

Add comment why error is not handled here